src/main/java/org/spongepowered/api/util/blockray/BlockRayHit.java

```diff
@@ -0,0 +1,141 @@
+package org.spongepowered.api.util.blockray;
+
+import com.flowpowered.math.GenericMath;
+import com.flowpowered.math.vector.Vector3d;
+import com.flowpowered.math.vector.Vector3i;
+import org.spongepowered.api.util.Direction;
+import org.spongepowered.api.world.extent.Extent;
+
+/**
+ * Represents a block hit by a ray. Stores more information than a regular location.
+ */
+public class BlockRayHit {
```

Comment:
I think there's enough information here to build a Location object, so a `toLocation()` method would be useful